src/Entity/Transformer.php

```diff
@@ -155,40 +204,51 @@ public function loadBlock(array $content, PageInterface $page): PageBlockInterfa
 
         $block->setPage($page);
 
-        if (isset($content['id'])) {
-            $block->setId($content['id']);
-        }
+        $blockClass = $this->blockManager->getClass();
+
+        // NEXT_MAJOR: Remove null support
+        if (null !== $this->serializer) {
```

Comment:
To avoid the "funky behavior", we should change this to
```suggestion
        if (null !== $this->serializer && method_exists('removeChild', $page) && method_exists('removeBlock', $page)) {
```